src/test/rustdoc/issue-43893.rs

```diff
@@ -0,0 +1,34 @@
+// Copyright 2016 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+// ignore-cross-compile
+
+#![crate_name = "foo"]
```

Comment:
Please remove all unneeded empty lines in here (like in the `impl` blocks).

Reply:
Which empty lines? Before and after the `// ignore-corss-compile`? They are just retained from `issue-34274.rs`, which served as a template. I don't think they should be removed (especially the first empty line).

> like in the `impl` blocks

Means "and likewise remove them in impl blocks" or "like you did in impl blocks"?

Reply:
Like this:

```rust
// Copyright 2016 The Rust Project Developers. See the COPYRIGHT
// file at the top-level directory of this distribution and at
// http://rust-lang.org/COPYRIGHT.
//
// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
// option. This file may not be copied, modified, or distributed
// except according to those terms.

// ignore-cross-compile

#![crate_name = "foo"]

pub trait SomeTrait {}
pub struct SomeStruct;

// @has foo/trait.SomeTrait.html '//a/@href' '../src/foo/issue-43893.rs.html#23-26'
impl SomeTrait for usize {}

// @has foo/trait.SomeTrait.html '//a/@href' '../src/foo/issue-43893.rs.html#29-32'
impl SomeTrait for SomeStruct {}
```

Reply:
> impl SomeTrait for usize {}
> {}

Other [src] link test (issue-34274.rs) tests for single line link. I thought it would be useful to also provide multiple line implementation, so that the link contains line range, not a single line.

Reply:
Having multi-line tests is useful to make sure that the link points to the full span.

Reply:
That's why I contracted one impl to one line, but left the other impl as three lines in the update.